### PR TITLE
Implement Delete Loan Command Fixes #31 and #46

### LINKED ISSUE #31: As a user who loans money regularly, I can delete a loan linked to a contact

### PR DESCRIPTION
**Pull request by xiaorui-ui**
Added Delete Loan Command and Delete Loan Parser classes. Modified LoanRecords class slightly to return size so as to support deletion of loans based on index.

Command format: `deleteloan INDEX l/LOAN_INDEX`
Command example: `deleteloan 1 l/1`

(Also edited gitignore to ignore bin folder in case of its accidental creation in vscode)

### CODE REVIEW

**Review by kjw142857 — CHANGES_REQUESTED** [review]
A couple of clarifications might be needed on the handling of searching for the person by name (do we want to search by index instead, to align with `LinkLoanCommand` and avoid errors when there are duplicate names)? Other than that, LGTM!

File: src/main/java/seedu/address/logic/commands/DeleteLoanCommand.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+ * Deletes a loan from the address book.
+ */
+public class DeleteLoanCommand extends Command {
+    public static final String COMMAND_WORD = "deleteloan";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Delete the specified loan number of the person identified"
+            + "by the index number used. "
```
[the remark this answers is not in the document]
Comment (xiaorui-ui) [reply src/main/java/seedu/address/logic/commands/DeleteLoanCommand.java:23]: Intended behaviour is by name, the description is unclear. I have changed the behaviour to by index as suggested

[the remark this answers is not in the document]
Comment (kjw142857) [reply src/main/java/seedu/address/logic/commands/DeleteLoanCommand.java]: Alright, thanks!

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 2]
```diff
@@ -10,9 +10,10 @@ public class CliSyntax {
     public static final Prefix PREFIX_PHONE = new Prefix("p/");
     public static final Prefix PREFIX_EMAIL = new Prefix("e/");
     public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
+    public static final Prefix PREFIX_INDEX = new Prefix("i/");
```
Comment (kjw142857) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:13]: Is this still necessary? Since we are now using PREFIX_LOAN_INDEX

**Review by kjw142857 — CHANGES_REQUESTED** [review]
Nice! Just 2 more very small things to fix (you can see my comments below).

**Review by kjw142857 — APPROVED** [review]
LGTM!

**Pull request merged**